src/Sarif.Converters/SemmleConverter.cs

```diff
@@ -11,8 +15,189 @@ namespace Microsoft.CodeAnalysis.Sarif.Converters
     /// </summary>
     internal class SemmleConverter : ToolFileConverterBase
     {
+        // Semmle logs are CSV files.
+        private static readonly string[] s_delimiters = new[] { "," };
+
+        // The fields are as follows:
+        private enum FieldIndex
+        {
+            QueryName,
+            QueryDescription,
+            Severity,
+            Message,
+            RelativePath,
+            Path,
+            StartLine,
+            StartColumn,
+            EndLine,
+            EndColumn
+        }
+
+        private TextFieldParser _parser;
+        private List<Notification> _toolNotifications;
+
+        /// <summary>
+        /// Converts a Semmle log file in CSV format to a SARIF log file.
+        /// </summary>
+        /// <param name="input">
+        /// Input stream from which to read the Semmle log.
+        /// </param>
+        /// <param name="output">
+        /// Output string to which to write the SARIF log.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when one or more required arguments are null.
+        /// </exception>
         public override void Convert(Stream input, IResultLogWriter output)
         {
+            if (input == null)
+            {
+                throw new ArgumentNullException(nameof(input));
+            }
+
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(output));
+            }
+
+            _toolNotifications = new List<Notification>();
+
+            var results = GetResultsFromStream(input);
+
+            var tool = new Tool
+            {
+                Name = "Semmle"
+            };
+
+            output.Initialize(id: null, correlationId: null);
+
+            output.WriteTool(tool);
+
+            output.OpenResults();
+            output.WriteResults(results);
+            output.CloseResults();
+
+            if (_toolNotifications.Any())
+            {
+                output.WriteToolNotifications(_toolNotifications);
+            }
+        }
+
+        private Result[] GetResultsFromStream(Stream input)
+        {
+            var results = new List<Result>();
+            using (_parser = new TextFieldParser(input)
+            {
+                TextFieldType = FieldType.Delimited,
+                Delimiters = s_delimiters
+            })
+            {
+                while (!_parser.EndOfData)
+                {
+                    results.Add(ParseResult());
+                }
+            }
+
+            return results.ToArray();
+        }
+
+        private Result ParseResult()
+        {
+            string[] fields = _parser.ReadFields();
+            return new Result
+            {
+                Level = ResultLevelFromSemmleSeverity(GetString(fields, FieldIndex.Severity)),
+                Message = fields[(int)FieldIndex.Message],
```

Comment:
>Message [](start = 16, length = 7)

Need to parse message here for encoded link in order to create related locations (and scrub link data from the message, we have no encoding strategy here). Probably, this scenario should be included in the SARIF todo around whether to allow for inlined formatting/etc. of messages (for bolding, encoding links, etc.)